subprojects/internal-integ-testing/src/main/groovy/org/gradle/test/fixtures/maven/M2Installation.groovy

```diff
@@ -98,8 +99,22 @@ class M2Installation implements Action<GradleExecuter> {
     void execute(GradleExecuter gradleExecuter) {
         init()
         gradleExecuter.withUserHomeDir(userHomeDir)
+        // if call `using m2`, then we disable the automatic isolation of m2
+        isolateMavenLocal = false
         if (globalMavenDirectory?.exists()) {
             gradleExecuter.withEnvironmentVars(M2_HOME: globalMavenDirectory.absolutePath)
         }
     }
+
+    void isolateMavenLocalRepo(GradleExecuter gradleExecuter) {
+        gradleExecuter.beforeExecute {
+            if (isolateMavenLocal) {
+                setMavenLocalLocation(gradleExecuter, temporaryFolder.testDirectory.createDir("m2-home-should-not-be-filled"))
```

Comment:
Can we make sure that this directory is empty after the test executed as we do now on CI? If we stop using the global home directory then the check we do now doesn't have any effect.

Reply:
We can, but I'd do it in a different PR as it may highlight more leakages (which is good) and I have no time to fix them.

Reply:
How can this highlight more leakages given that we check for a clean M2 directory on CI already?

Reply:
Ok you trust our infra a lot more than I do :)

Reply:
Added